code/backend/batch/utilities/tools/QuestionAnswerTool.py

```diff
@@ -1,182 +1,113 @@
+        # Now that we are querying Azure Search using the official libraries, we can add and modify the query any way we need, such as adding a query for image embeddings
+        vector_query = VectorizedQuery(
+            vector=question_embeddings,
+            k_nearest_neighbors=self.env_helper.AZURE_SEARCH_TOP_K,
+            fields="content_vector",
+        )
+        search_results = self.azure_search_helper.search_client.search(
+            search_text=question,
+            top=self.env_helper.AZURE_SEARCH_TOP_K,
+            vector_queries=[vector_query],
         )
 
-        return answering_prompt, {
-            "sources": sources_text,
-            "question": question,
-        }
-
-    def generate_on_your_data_llm_chain(
-        self,
-        question: str,
-        chat_history: list[dict],
-        sources: list[Document],
-    ):
-        examples = []
-
-        few_shot_example = {
-            "sources": self.config.example.documents.strip(),
-            "question": self.config.example.user_question.strip(),
-            "answer": self.config.example.answer.strip(),
-        }
-
-        if few_shot_example["sources"]:
-            few_shot_example["sources"] = QuestionAnswerTool.json_remove_whitespace(
-                few_shot_example["sources"]
-            )
-
-        if any(few_shot_example.values()):
-            if all((few_shot_example.values())):
-                examples.append(few_shot_example)
-            else:
-                warnings.warn(
-                    "Not all example fields are set in the config. Skipping few-shot example."
+        retrieved_documents = []
+        source_documents = []
+        i = 1
+        for page in search_results.by_page():
+            for document in page:
+                retrieved_documents.append(
+                    {f"[doc{i}]": {"content": document.get("content")}}
                 )
+                source_documents.append(
+                    SourceDocument(
+                        id=document.get("id"),
+                        content=document.get("content"),
+                        title=document.get("title"),
+                        source=document.get("source"),
+                        chunk=document.get("chunk"),
+                        offset=document.get("offset"),
+                        page_number=document.get("page_number"),
+                    )
+                )
+                i += 1
 
-        example_prompt = ChatPromptTemplate.from_messages(
-            [
-                HumanMessagePromptTemplate.from_template(
-                    self.config.prompts.answering_user_prompt
-                ),
-                AIMessagePromptTemplate.from_template("{answer}"),
-            ]
-        )
-
-        few_shot_prompt = FewShotChatMessagePromptTemplate(
-            example_prompt=example_prompt,
-            examples=examples,
+        ret_docs_json = json.dumps(
```

Comment:
The JSON sources is only used with the new format, but a different sources format is used with the old format, which is why it was separate before